addons/website/static/src/services/website_service.js

```diff
@@ -46,6 +46,7 @@ export const websiteService = {
             snippetsLoaded: false,
             isMobile: false,
             showPageProperties: false,
+            isTestMode: odoo.debug.includes('tests'),
```

Comment:
Hello, I think technically `isTestMode` does not need to be in the context object, which is reactive to rerender components when one of its key is changed (as the webclient needs to be reloaded to add/remove tests). 
So maybe it could even be defined as a property of the client action in the setup ?
Overall I think it's way better to have clearly defined steps, whereas before we didn't even know which part of "starting the webclient on the client action in edit mode" would take time and make the timeout fail...

Reply:
You are right but on top of that, it wasn't even working properly, being in test mode is not only about being in `debug=tests` but is also/or about being in test mode python wise.
Not sure how I can forgot about that since I am the one who made it but.. :grimacing: https://github.com/odoo/odoo/commit/33e0acb2acc41ada3351522ba7dfdd149d354d11

Anyway, I refactored that part of the PR, I don't think there is a clean and existing way to know if we are in test mode, so I made my own for the website preview.
It seems to be a very niche need as I could not find in community such a thing. Arguably it could be something the webclient provide, but given what I explained, I think having that only available in the website preview client action is fine (for now).

I'll let you check the new implementation :+1: